apps/condo/domains/billing/schema/AllResidentBillingReceiptsService.test.js

```diff
@@ -43,6 +45,73 @@ describe('AllResidentBillingReceipts', () => {
         expect(objs[0].id).toEqual(receipt.id)
     })
 
+    test('user with valid serviceAccount can filter residentBillingReceipts by serviceConsumer', async () => {
+        // User has flats in building A and building B
+        // Each building has own BillingOrganizationIntegrationContext
+        // User is able to get receipts for both of his buildings
+
+        const userClient = await makeClientWithProperty()
+        const adminClient = await makeLoggedInAdminClient()
+
+        // User has two flats in building A:
+        const [integrationA] = await createTestBillingIntegration(adminClient)
+        const [contextA] = await createTestBillingIntegrationOrganizationContext(adminClient, userClient.organization, integrationA)
+        const [billingPropertyA] = await createTestBillingProperty(adminClient, contextA)
+        const [billingAccountA, billingAccountAttrsA] = await createTestBillingAccount(adminClient, contextA, billingPropertyA)
+        const [billingAccountA2, billingAccountAttrsA2] = await createTestBillingAccount(adminClient, contextA, billingPropertyA)
+
+        await addResidentAccess(userClient.user)
+        const [residentA] = await createTestResident(adminClient, userClient.user, userClient.organization, userClient.property, {
+            unitName: billingAccountAttrsA.unitName,
+        })
+
+        const serviceConsumerPayloadA = {
+            residentId: residentA.id,
+            unitName: billingAccountAttrsA.unitName,
+            accountNumber: billingAccountAttrsA.number,
+        }
+        const [serviceConsumerA] = await registerServiceConsumerByTestClient(userClient, serviceConsumerPayloadA)
+
+        const serviceConsumerPayloadA2 = {
+            residentId: residentA.id,
+            unitName: billingAccountAttrsA2.unitName,
+            accountNumber: billingAccountAttrsA2.number,
+        }
+        const [serviceConsumerA2] = await registerServiceConsumerByTestClient(userClient, serviceConsumerPayloadA2)
```

Comment:
Why are you unwrapping vars, if you're not using it later?

Reply:
I'd done it for debug purposes :(